spinnman/spalloc/spalloc_job.py

```diff
@@ -211,6 +212,42 @@ def get_session_credentials_for_db(self) -> Mapping[Tuple[str, str], str]:
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def write_data(self, x: int, y: int, address: int, data: bytes) -> None:
+        """
+        Write data to a given address on a given chip of the job.
+
+        :param int x: The X coordinate of the chip
+        :param int y: The Y coordinate of the chip
+        :param int address: The address to write to
+        :param bytes data: The data to write
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_data(self, x: int, y: int, address: int, size: int) -> bytes:
+        """
+        Write data to a given address on a given chip of the job.
+
+        :param int x: The X coordinate of the chip
+        :param int y: The Y coordinate of the chip
+        :param int address: The address to write to
+        :param int size: The number of bytes to read
+        :return: The data read
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def prepare_routers(
```

Comment:
if Java names change here too

Reply:
prepare_routing_tables is actually an existing Transceiver method name for this functionality